source/core/ut_metadata.pkb

```diff
@@ -327,12 +283,16 @@ create or replace package body ut_metadata as
   end; 
     
   function get_object_name(a_full_object_name in varchar2) return varchar2 is
-    l_schema varchar2(250);
-    l_object varchar2(250);
-    l_procedure_name varchar2(250);
+    l_result varchar2(250);
   begin
-    ut_metadata.do_resolve(a_full_object_name,7,l_schema,l_object, l_procedure_name);
-    return l_object;
+    l_result := regexp_substr(
+      a_full_object_name,
+      '^([A-Za-z0-9$#_]+|".*?")\.([A-Za-z0-9$#_]+|".*?")', subexpression => 2
+      );
+    if not l_result like '"%"' then
+      l_result := upper(l_result);
+    end if;
+    return l_result;
```

Comment:
I would do here dbms_assert to verify that we have a correct name returned in case regexp result got name not right.

Reply:
good catch